dot/parachain/dispute/types/vote.go

```diff
@@ -0,0 +1,291 @@
+	if ok {
+		return false
+	}
+
+	voted, ok := val.(Voted)
+	if !ok {
+		return false
+	}
+
+	return len(voted.Votes) == 0
+}
+
+// ApprovalVotes returns the approval votes for the candidate
+func (v *OwnVoteState) ApprovalVotes() ([]Vote, error) {
+	vdt := scale.VaryingDataType(*v)
+	val, err := vdt.Value()
+	if err != nil {
+		return nil, fmt.Errorf("getting value from OwnVoteState vdt: %w", err)
+	}
+
+	_, ok := val.(CannotVote)
+	if ok {
+		return nil, nil
+	}
+
+	voted, ok := val.(Voted)
+	if !ok {
+		return nil, fmt.Errorf("invalid type for OwnVoteState: expected Voted, got %T", val)
+	}
+
+	var votes []Vote
+	for _, vote := range voted.Votes {
+		disputeStatement, err := vote.DisputeStatement.Value()
+		if err != nil {
+			return nil, fmt.Errorf("getting value from DisputeStatement vdt: %w", err)
+		}
+
+		_, ok := disputeStatement.(inherents.ApprovalChecking)
+		if !ok {
+			continue
+		}
+
+		votes = append(votes, Vote{
+			ValidatorIndex:     vote.ValidatorIndex,
+			ValidatorSignature: vote.ValidatorSignature,
+			DisputeStatement:   vote.DisputeStatement,
+		})
+	}
+
+	return votes, nil
+}
+
+// Votes returns the votes for the candidate
+func (v *OwnVoteState) Votes() ([]Vote, error) {
+	vdt := scale.VaryingDataType(*v)
+	val, err := vdt.Value()
+	if err != nil {
+		return nil, fmt.Errorf("getting value from OwnVoteState vdt: %w", err)
+	}
+
+	_, ok := val.(CannotVote)
+	if ok {
+		return nil, nil
+	}
+
+	voted, ok := val.(Voted)
+	if !ok {
+		return nil, fmt.Errorf("invalid type for OwnVoteState: expected Voted, got %T", val)
+	}
+
+	return voted.Votes, nil
+}
+
+// NewOwnVoteState returns a new OwnVoteState with the given value
+func NewOwnVoteState(value scale.VaryingDataTypeValue) (OwnVoteState, error) {
+	vdt, err := scale.NewVaryingDataType(Voted{}, CannotVote{})
+	if err != nil {
+		return OwnVoteState{}, fmt.Errorf("creating new OwnVoteState vdt: %w", err)
+	}
+
+	err = vdt.Set(value)
+	if err != nil {
+		return OwnVoteState{}, fmt.Errorf("setting value to OwnVoteState vdt: %w", err)
+	}
+
+	return OwnVoteState(vdt), nil
+}
+
+// CandidateVoteState is the state of the votes for a candidate
+type CandidateVoteState struct {
+	Votes         CandidateVotes
+	Own           OwnVoteState
+	DisputeStatus *DisputeStatus
+}
+
+func (c *CandidateVoteState) IsDisputed() bool {
+	return c.DisputeStatus != nil
+}
+
+func (c *CandidateVoteState) IsConfirmed() (bool, error) {
+	if c.DisputeStatus == nil {
+		return false, nil
+	}
+
+	return c.DisputeStatus.IsConfirmedConcluded()
+}
+
+func (c *CandidateVoteState) IsConcludedFor() (bool, error) {
+	if c.DisputeStatus == nil {
+		return false, nil
+	}
+
+	return c.DisputeStatus.IsConcludedFor()
+}
+
+func (c *CandidateVoteState) IsConcludedAgainst() (bool, error) {
+	if c.DisputeStatus == nil {
+		return false, nil
+	}
+
+	return c.DisputeStatus.IsConcludedAgainst()
+}
+
+// NewCandidateVoteState creates a new CandidateVoteState
+// TODO: implement this later since nothing is using it yet
```

Comment:
Should we have an issue for this?

Reply:
I've already implemented it in `feat/k/disputes-coordinator`. Also, this is just going into another feature branch. I don't think we need an issue for this.

Reply:
sounds good to me!